nexus/types/src/inventory.rs

```diff
@@ -134,6 +134,14 @@ impl Collection {
     ) -> impl Iterator<Item = &OmicronZoneConfig> {
         self.omicron_zones.values().flat_map(|z| z.zones.zones.iter())
     }
+
+    /// Iterate over the sled ids of sleds identified as Scrimlets
+    pub fn scrimlets(&self) -> impl Iterator<Item = Uuid> + '_ {
+        self.sled_agents
+            .iter()
+            .filter(|(_, inventory)| inventory.sled_role == SledRole::Scrimlet)
+            .map(|(sled_id, _)| *sled_id)
```

Comment:
Tiny nit - if you want to squish this down to `filter_map`, you could use https://doc.rust-lang.org/std/primitive.bool.html#method.then_some (which I only learned about recently):

```rust
.filter_map(|(sled_id, inventory)| (inventory.sled_role == SledRole::Scrimlet).then_some(*sled_id))

Reply:
This is neat and I didn't know it.  But I think I like what's here better in this case.